tracing/opentracing/grpc.go

```diff
@@ -37,12 +38,15 @@ func ToGRPCRequest(tracer opentracing.Tracer, logger log.Logger) func(ctx contex
 // The logger is used to report errors and may be nil.
 func FromGRPCRequest(tracer opentracing.Tracer, operationName string, logger log.Logger) func(ctx context.Context, md *metadata.MD) context.Context {
 	return func(ctx context.Context, md *metadata.MD) context.Context {
-		span, err := tracer.Join(operationName, opentracing.TextMap, metadataReaderWriter{md})
+		var span opentracing.Span
+		wireContext, err := tracer.Extract(opentracing.TextMap, metadataReaderWriter{md})
 		if err != nil {
+			logger.Log("err", err)
+		}
+		if wireContext != nil {
+			span = tracer.StartSpan(operationName, ext.RPCServerOption(wireContext))
+		} else {
 			span = tracer.StartSpan(operationName)
```

Comment:
I guess (?) we could include the rpc server tag here even if there's no client SpanContext to reference...